app/services/doc_auth/socure/requests/document_request.rb

```diff
@@ -47,6 +47,40 @@ def handle_http_response(http_response)
           JSON.parse(http_response.body, symbolize_names: true)
         end
 
+        def handle_invalid_response(http_response)
+          message = [
+            self.class.name,
+            'Unexpected HTTP response',
+            http_response&.status,
+          ].join(' ')
+          exception = DocAuth::RequestError.new(message, http_response&.status)
+
+          response_body = begin
+            http_response&.body.present? ? JSON.parse(http_response.body) : {}
+          rescue JSON::JSONError
+            {}
+          end
+          handle_connection_error(
+            exception: exception,
+            status: response_body.dig('status'),
+            status_message: response_body.dig('msg'),
+          )
+        end
```

Comment:
it seems that this method is identical in DocumentRequest and DocVResultRequest. instead of having duplicate methods, this method should should be removed form DocumentRequest and DocVResultRequest, and moved to DocAuth::Socure::Request